### ISSUE TITLE
Failed to start quick debugging

### ISSUE THREAD

**Issue by gvcallen** [microsoft/vscode-cmake-tools#1310]
### Brief Issue Summary

I am unable to get quick debugging working, and the error message from the C++ extension appears when I hit Ctrl + F5. I am able to configure and build correctly using F7. I would like to use quick debugging as setting up a launch.json file for each project is not necessary for me.

### Expected:

1. Press Ctrl + F5
2. Quick debugging launches with relevant debugger

### Apparent Behavior:

1. Press Ctrl + F5
2. Error message from C++ extension appears:
"Unable to start debugging. Launch options string provided by the project system is invalid. Unable to determine path to debugger. Please specifiy the "MIDebuggerPath" option."

### Platform and Versions

- **Operating System**:  Windows
- **CMake Version**: 3.17.1
- **VSCode Version**: Latest as of 15/06/2020
- **CMake Tools Extension Version**: 1.4.0
- **Compiler/Toolchain**: x86_64-w64-mingw32 10.1.0

**Comment by andreeis** [comment]
I reproduce this and it looks like is not a regression. We need to fix some regexp and logic in the debugger detection logic.
This happens with any kit that points to a compiler that has a different naming convention than the extension expects. For example, x86_64-pc-msys-g++.exe instead of g++.

You can get unblocked easily by defining miDebuggerPath in settings, like below:
    "cmake.debugConfig": {
        "miDebuggerPath": "path to gdb",
    }

**Comment by gvcallen** [comment]
The issue is that I have two kits which I use interchangeably, specifically the MinGW-w64 32-bit and 64-bit toolchains from http://winlibs.com/. The GDB debugger for 64-bit is incapable of debugging 32-bit targets on Windows for now and of course vica-versa is impossible too, and so I have to use the specifically provided debugger for each kit. Of course I can setup multiple launch configurations, but that makes everything more manual and more complicated.

I've managed to "hack" it now by using the kit's name variable as the location for gdb, but ideally I would like to use variable substitution for any environment variable in my kits to make the launch.json simple get the "debuggerPath" from the kit (I have requested this here: https://github.com/microsoft/vscode-cmake-tools/issues/1312).

I would like to use CMake Tools for scalable C++ projects, but things like this are unfortunately holding me back. Otherwise everything else is great.
